.circleci/deploy-ghpages.sh

```diff
@@ -0,0 +1,48 @@
+#!/bin/sh
+# ideas used from https://gist.github.com/motemen/8595451
+
+# Based on https://github.com/eldarlabs/ghpages-deploy-script/blob/master/scripts/deploy-ghpages.sh
+# Used with their MIT license https://github.com/eldarlabs/ghpages-deploy-script/blob/master/LICENSE
+
+# abort the script if there is a non-zero error
+set -e
+
+# show where we are on the machine
+pwd
+remote=$(git config remote.origin.url)
+
+# make a directory to put the gp-pages branch
+mkdir gh-pages-branch
+cd gh-pages-branch
+# now lets setup a new repo so we can update the gh-pages branch
+git config --global user.email "$GH_EMAIL" > /dev/null 2>&1
+git config --global user.name "$GH_NAME" > /dev/null 2>&1
+git init
+git remote add --fetch origin "$remote"
+
+# switch into the the gh-pages branch
+if git rev-parse --verify origin/gh-pages > /dev/null 2>&1
+then
+    git checkout gh-pages
+    # delete any old site as we are going to replace it
+    # Note: this explodes if there aren't any, so moving it here for now
+    git rm -rf .
+else
+    git checkout --orphan gh-pages
+fi
```

Comment:
if

Reply:
fi terminates the preceding if in shell scripting.